src/Microsoft.CodeAnalysis.Razor.Workspaces/WorkspaceMonitorProvider.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public class WorkspaceMonitorProvider
```

Comment:
I'd suggest to tying the monitor to a single workspace, rather than keeping a global 'workspace tracker'.